dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -24,8 +23,18 @@
   {% endif %}
 {% endmacro %}
 
+{% macro dbt_snowflake_create_temp_relation(strategy, tmp_relation, sql) %}
```

Comment:
I spent a few minutes thinking about whether it makes sense to combine `dbt_snowflake_get_incremental_sql` + `dbt_snowflake_create_temp_relation`, to avoid writing the same condition check for `strategy` twice.

In order to make that happen, we'd need to wrap these materialization steps into a third macro that gets called by both. Something like:
```sql
{% macro prep_for_incremental_sql(on_schema_change, tmp_relation, target_relation) %}
    {% do adapter.expand_target_column_types(
           from_relation=tmp_relation,
           to_relation=target_relation) %}
    {#-- Process schema changes. Returns dict of changes if successful. Use source columns for upserting/merging --#}
    {% set dest_columns = process_schema_changes(on_schema_change, tmp_relation, target_relation) %}
    {% if not dest_columns %}
      {% set dest_columns = adapter.get_columns_in_relation(target_relation) %}
    {% endif %}
    {{ return(dest_columns) }}
{% endmacro %}
```
Then we have a macro that does:
```sql
{% macro dbt_snowflake_get_incremental_sql(strategy, tmp_relation, target_relation, unique_key, dest_columns) %}
  {% if strategy == 'merge' %}
    {% do run_query(create_view_as(tmp_relation, sql)) %}
    {% set dest_columns = prep_for_incremental_sql(on_schema_change, tmp_relation, existing_relation) %}
    {% do return(get_merge_sql(target_relation, tmp_relation, unique_key, dest_columns)) %}
  {% elif strategy == 'delete+insert' %}
    {% do run_query(create_table_as(True, tmp_relation, sql)) %}
    {% set dest_columns = prep_for_incremental_sql(on_schema_change, tmp_relation, existing_relation) %}
    {% do return(get_delete_insert_merge_sql(target_relation, tmp_relation, unique_key, dest_columns)) %}
  {% else %}
    {% do exceptions.raise_compiler_error('invalid strategy: ' ~ strategy) %}
  {% endif %}
{% endmacro %}
```
And finally, in the materialization itself:
```sql
{% else %}
  {% set build_sql = dbt_snowflake_get_incremental_sql(strategy, tmp_relation, target_relation, unique_key, dest_columns) %}
{% endif %}
```
That might be totally overkill! Or it might leave the materialization a bit leaner + more modular